src/widgets/radio.jsx

```diff
@@ -574,17 +643,32 @@ var RadioEditor = React.createClass({
 
     onDelete: function(choiceIndex, e) {
         e.preventDefault();
+
         var choices = this.props.choices.slice();
+        var deleted = choices[choiceIndex];
+
         choices.splice(choiceIndex, 1);
-        this.props.onChange({choices: choices});
+
+        this.props.onChange({
+            choices: choices,
+            hasNoneOfTheAbove: !( deleted.isNoneOfTheAbove || !this.props.hasNoneOfTheAbove )
```

Comment:
Yeah I think after I while I got tangled in finding the correct statement and missed the more elegant solution!